### ISSUE TITLE
Collisions with Dynamic Trimesh

### ISSUE THREAD

**Issue by karthikmahesh** [schteppe/cannon.js#191]
I am pretty new to Cannon, and I still have difficulty understanding the way collision works. My application requires me to use the undocumented Trimesh object3D structure. One such trimesh object is generated on the fly for each frame and is required to collide/ contact with a static rigid box. I have setup parameters for collision, similar to examples in the documentation for other types of object3D bodies.

world.defaultContactMaterial.contactEquationStiffness = 1e9;
world.defaultContactMaterial.contactEquationRelaxation = 4;
var physicsMaterial = new CANNON.Material("slipperyMaterial");
var physicsContactMaterial = new CANNON.ContactMaterial(physicsMaterial, 														physicsMaterial, 0.0, 0.3);
world.addContactMaterial(physicsContactMaterial);															
objectBody.linearDamping = 0.1;
objectBody.velocity.set(0.1,0,0.1);
objectBody.angularVelocity.set(0,0,0);	

I have also setup nominal linear velocities for the dynamic body, that are updated each frame with trimesh recalculation, while the other object is static. They also have equal masses of 0.5 (also tried higher values). During the collision the dynamic trimesh just passes through the static body, without moving it. Can you possibly let me know what parameters I would need to verify and what additional collision/ intersection/ contact calls that I would need to make, in order to ensure that the objects dont pass through each other, but get moved as a result of motion of the dynamic trimesh body.
Thanks

**Comment by karthikmahesh** [comment]
Also, number of iterations is kept high for precise calculations

**Comment by schteppe** [comment]
Hi,
Trimesh vs Box collision is not implemented yet. How about using a sphere instead? Or 8 compound spheres?

**Comment by karthikmahesh** [comment]
I am sorry, but do you mean Trimesh vs Sphere collision? or approximating the trimesh with spheres? any pointers to do this will be helpful

**Comment by schteppe** [comment]
Sorry. I meant approximating the box with spheres. In that case they would collide. Approximating the trimesh with spheres would work too.

Is the trimesh vertices changing over time?

**Comment by karthikmahesh** [comment]
Yes, thanks for the info. The trimesh is too complex and is also changing over time. I will update this post on how approximating the other object works out

**Comment by schteppe** [comment]
It's been a while, how's it going with that mesh?
